polynote-runtime/src/main/scala/polynote/runtime/ReprsOf.scala

```diff
@@ -98,6 +98,12 @@ private[runtime] trait CollectionReprs extends FromDataReprs { self: ReprsOf.typ
       Array(repr)
   }
 
+  implicit def map[F[KK, VV] <: Map[KK, VV], K, V](implicit structEncoder: DataEncoder.StructDataEncoder[(K, V)]): ReprsOf[F[K, V]] = instance {
+    map =>
+      Array(StreamingDataRepr.fromHandle(new StructSeqStreamHandle[(K,V), (K,V)](_, map.toSeq, identity, structEncoder)))
+  }
```

Comment:
Hope I did this right! 😬

Reply:
Seems OK to me, though it won't help with map fields of case classes (need a `DataEncoder` for that). Maybe we should introduce a new `DataType` for maps?

Reply:
I ended up removing this and making a DataEncoder instead and introduced a new MapType per your suggestions :)